MediaSession.cpp

```diff
@@ -231,6 +233,52 @@ std::string MediaKeySession::KeyIdsToJSON() {
 }
 
 bool MediaKeySession::ParseClearKeyInitializationData(const std::string& initData, std::string* output)
+{
+   bool result = false;
+   const char identifier[] = { '"', 'k', 'i', 'd', 's', '"', ':', '\0' };
+
+   /* keyids type */
+   if(initData.find(identifier) != std::string::npos) {
+       result = ParseKeyIdsInitData(initData, output) > 0 ? true : false;
+   }
+   else {
+       result = ParseCENCInitData(initData, output) > 0 ? true : false;
+   }
+   return result;
+}
+
+using JSONStringArray = Core::JSON::ArrayType<Core::JSON::String>;
+
+bool MediaKeySession::ParseKeyIdsInitData(const std::string& initData, std::string* output) {
+    class InitData : public Core::JSON::Container {
+    public:
+        InitData() : Core::JSON::Container() , KeyIds() {
+            Add(_T("kids"), &KeyIds);
+        }
+        virtual ~InitData() {
+        }
+
+    public:
+        JSONStringArray KeyIds;
+    } jsonData;
+
+    output->clear();
+
+    jsonData.FromString(initData);
+    JSONStringArray::ConstIterator index(static_cast<const InitData&>(jsonData).KeyIds.Elements());
+
+    cout << "Clearkey CDMi: initdata keyid(s): [";
```

Comment:
Can we make these TRACE_L1 statements, so they only show up in DEBUG ? This will clutter up other messages if they are on by default.

Reply:
As std::cout is used all over the file, same implemented here also. Yes, TRACE_L1 is the good approach to show up only in DEBUG. Let me change now here and later can raise another pull request to change it all over the file.